src/QualificationRunner.Core/Services/QualificationRunner.cs

```diff
@@ -163,22 +163,23 @@ private async Task<ObservedDataMapping> copyObservedData(ObservedDataMapping obs
          };
       }
 
-      private Task<Section[]> copySectionContents(Section[] sections)
+      private Section[] copySectionContents(Section[] sections)
       {
+         // Not in parallel for now to ensure that we do not override files already downloaded as we are checking for file existence
          if (sections == null)
-            return Task.FromResult(Array.Empty<Section>());
+            return Array.Empty<Section>();
 
-         return Task.WhenAll(sections.Select(copySectionContent));
+         return sections.Select(copySectionContent).ToArray();
       }
 
-      private async Task<Section> copySectionContent(Section section)
+      private Section copySectionContent(Section section)
       {
          Task<string> downloadRemoteSectionContent() => downloadRemoteFile(section.Content, CONTENT_DOWNLOAD_FOLDER, "Content");
 
          var contentAbsolutePath = absolutePathFrom(_runOptions.ConfigurationFolder, section.Content);
 
          if (!localFileExists(contentAbsolutePath))
-            contentAbsolutePath = await downloadRemoteSectionContent();
+            contentAbsolutePath = downloadRemoteSectionContent().Result;

```

Comment:
Not great . Idealy the downloadRemoteFile would lock on the file name to check if it exists before downloading it but well..too much work for not much gain